control-plane/connect-inject/endpoints_controller.go

```diff
@@ -1004,10 +1004,22 @@ func (r *EndpointsController) consulNamespace(namespace string) string {
 
 // hasBeenInjected checks the value of the status annotation and returns true if the Pod has been injected.
 func hasBeenInjected(pod corev1.Pod) bool {
-	if anno, ok := pod.Annotations[keyInjectStatus]; ok {
-		if anno == injected {
-			return true
-		}
+	if anno, ok := pod.Annotations[keyInjectStatus]; ok && anno == injected {
+		return true
 	}
 	return false
 }
+
```

Comment:
This function doesn't appear to be used by anyone other than the place it was removed from.
Would probably make sense to remove this change and its tests, unless there was some reason that I'm not understanding. Was there a bug that is fixed in this?

Reply:
I think this was added for readability.